### PR TITLE
make some fields required for TGenotype, TAltAllele, TVariant, TLocus…

### PR DESCRIPTION
**Pull request by catoverdrive**
…, TInterval. also MatrixType.rowType.gs

### CODE REVIEW

File: src/main/scala/is/hail/expr/Type.scala [hunk 1]
```diff
@@ -1249,7 +1249,7 @@ object TGenotype {
   def representation(required: Boolean = false): TStruct = {
     val t = TStruct(
       "gt" -> TInt32(),
-      "ad" -> TArray(TInt32()),
+      "ad" -> TArray(!TInt32()),
       "dp" -> TInt32(),
       "gq" -> TInt32(),
       "pl" -> TArray(TInt32()))
```
Comment (cseed) [inline src/main/scala/is/hail/expr/Type.scala]: Not pl?  pl!  pl!

Comment (catoverdrive) [reply src/main/scala/is/hail/expr/Type.scala]: oops.

**Comment by catoverdrive** [comment]
I... think I just forgot to push those commits.

**Comment by cseed** [comment]
This is failing because AD and PL arrays are (properly, I would argue) being dropped by buildGenotypeExtractor because their elements aren't required.  This won't work unless  import_vcf sets requiredness on import.  Here is a possible diff (which is probably subsumed by your next PR) which fixes some (but maybe not all) of the failing tests:

```
diff --git a/src/main/scala/is/hail/io/vcf/LoadVCF.scala b/src/main/scala/is/hail/io/vcf/LoadVCF.scala
index 77f6c72..fadc936 100644
--- a/src/main/scala/is/hail/io/vcf/LoadVCF.scala
+++ b/src/main/scala/is/hail/io/vcf/LoadVCF.scala
@@ -109,7 +109,7 @@ object LoadVCF {
         (line.getType == VCFHeaderLineType.Flag && line.getCount == 0)))
       Field(id, baseType, i, attrs)
     else
-      Field(id, TArray(baseType), i, attrs)
+      Field(id, TArray(!baseType), i, attrs)
   }
 
   def headerSignature[T <: VCFCompoundHeaderLine](lines: java.util.Collection[T],
@@ -124,10 +124,10 @@ object LoadVCF {
     callFields: Set[String] = Set.empty[String]): (TStruct, Int) = {
     val canonicalFields = Array(
       "GT" -> TCall(),
-      "AD" -> TArray(TInt32()),
+      "AD" -> TArray(!TInt32()),
       "DP" -> TInt32(),
       "GQ" -> TInt32(),
-      "PL" -> TArray(TInt32()))
+      "PL" -> TArray(!TInt32()))
 
     val raw = headerSignature(lines, callFields)
 
```

**Comment by catoverdrive** [comment]
moved changes into #2417.

**Pull request closed without merging**